### ISSUE TITLE
eslint exhaustive-deps incorrectly labelling constant value as key dependency

### ISSUE THREAD

**Issue by samMatenaer** [TanStack/query#5196]
### Describe the bug

`@tanstack/eslint-plugin-query@4.27.0`

incorrect exhaustive-deps error for constant values when using them in the queryFn
```ts
const CONST_VAL = 1;

const useFoo = () => {

  // queryKey eslint error: The following dependencies are missing in your queryKey: CONST_VAL
  const testQuery = useQuery({
    queryKey: ['foo'],
    queryFn: () => CONST_VAL,
  });

  // no react-hooks error when using const
  useEffect(() => {
    const newVal = 1 + CONST_VAL;
  }, []);

};
```

### Steps to reproduce

use a constant in the `queryFn` property for a `useQuery` hook. then you will see an error saying that the constant is not in the dependency array

### Expected behavior

constants should not count as a dependency

### Platform

vscode

### Tanstack Query adapter

react-query

### TanStack Query version

4.27.0

**Comment by TkDodo** [comment]
can confirm, here is a failing test case:

```
    {
      name: 'should ignore constants defined out of scope',
      code: `
        const CONST_VAL = 1
        function MyComponent() {
            useQuery({
              queryKey: ["foo"],
              queryFn: () => CONST_VAL
            });
        }
      `,
    },
```

@Newbie012 FYI

**Comment by Newbie012** [comment]
Hopefully I'll have the time today to fix this and the other issue.

**Comment by Newbie012** [comment]
Should be fixed once https://github.com/TanStack/query/pull/5217 is merged.

**Comment by am-maneaters** [comment]
@TkDodo @Newbie012 the new update fixes this issue when the `useQuery` is called from within a `function Foo(){}` definition but not a `const Foo = () => {}` arrow function (i.e. my original example). see additional test case below

```
    {
      name: 'should ignore constants defined out of arrow function scope',
      code: `
        const CONST_VAL = 1
        const MyComponent = () => {
            useQuery({
              queryKey: ["foo"],
              queryFn: () => CONST_VAL
            });
        }
      `,
    },
```

**Comment by luizeboli** [comment]
@TkDodo Just saw its supposed to be fixed by the latest release, however I'm still experiencing this error when using an external function to define the query:

```js
const Ab = 1

// This throws the error
function groupsQuery() {
  return {
    queryKey: ['get-groups'],
    queryFn: () => Ab,
  }
}

// This works
export function useGroups() {
  const Ac = 1
  return useQuery({
    queryKey: ['get-groups'],
    queryFn: () => Ac,
  })
}
```

**Comment by TkDodo** [comment]
please create a new issue for this @luizeboli
